src/dos/drive_virtual.cpp

```diff
@@ -198,22 +211,20 @@ void VFILE_Register(const char *name, const std::vector<uint8_t> &blob, const ch
 	VFILE_Register(name, blob.data(), check_cast<uint32_t>(blob.size()), dir);
 }
 
-void VFILE_Remove(const char *name, const char *dir = "")
+void VFILE_Update(const char* name, const std::vector<uint8_t>& blob, const char* dir)
 {
-	assert(name);
-	assert(dir);
-	unsigned int onpos = 0;
-	if (*dir) {
-		for (unsigned int i = 1; i < vfile_pos; i++)
-			if (!strcasecmp(vfilenames[i].shortname.c_str(), dir) ||
-			    !strcasecmp(vfilenames[i].fullname.c_str(), dir)) {
-				onpos = i;
-				break;
-			}
-		if (onpos == 0)
-			return;
+	auto vfile = find_vfile_by_name_and_dir(name, dir);
+
+	if (vfile) {
+		vfile->data    = std::make_shared<std::vector<uint8_t>>();
+		*(vfile->data) = blob;
 	}
-	auto vfile = find_vfile_by_name_and_pos(name, onpos);
+}
```

Comment:
Suggest taking in: `std::vector<uint8_t> blob`

Then you `blob` can be moved directly into the new shared vector:

```c++
if (vfile) {
    vfile->data = std::make_shared<std::vector<uint8_t>>(std::move(blob));
}
```

(see my other comment in autoexec.bat on the calling side also moving the vector into the function)

Reply:
OK, changed.